mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -629,6 +637,10 @@ OpFoldResult gpu::SubgroupReduceOp::fold(FoldAdaptor /*adaptor*/) {
     return getResult();
   }
 
+  if (getClusterSize() == 1) {
+    return getValue();
+  }
```

Comment:
Wouldn't it be better to check for this before uniformity? This seems like a cheaper way of doing things because we won't ever have to fold twice.